include/manif/impl/bundle/BundleTangent_base.h

```diff
@@ -0,0 +1,398 @@
+#ifndef _MANIF_MANIF_BUNDLETANGENT_BASE_H_
+#define _MANIF_MANIF_BUNDLETANGENT_BASE_H_
+
+#include "manif/impl/bundle/Bundle_properties.h"
+#include "manif/impl/tangent_base.h"
+
+namespace manif
+{
+
+/**
+ * @brief The base class of the Bundle tangent.
+ */
+template<typename _Derived>
+struct BundleTangentBase : TangentBase<_Derived>
+{
+private:
+  using Base = TangentBase<_Derived>;
+  using Type = BundleTangentBase<_Derived>;
+
+  using IdxList = typename internal::traits<_Derived>::IdxList;
+
+  using BegDoF = typename internal::traits<_Derived>::BegDoF;
+  using LenDoF = typename internal::traits<_Derived>::LenDoF;
+
+  using BegRep = typename internal::traits<_Derived>::BegRep;
+  using LenRep = typename internal::traits<_Derived>::LenRep;
+
+  using BegAlg = typename internal::traits<_Derived>::BegAlg;
+  using LenAlg = typename internal::traits<_Derived>::LenAlg;
+
+  template<int _Idx>
+  using PartType = typename internal::traits<_Derived>::template PartType<_Idx>;
+
+public:
+  MANIF_TANGENT_TYPEDEF
+  MANIF_INHERIT_TANGENT_API
+  MANIF_INHERIT_TANGENT_OPERATOR
+
+  using Base::data;
+  using Base::coeffs;
+
+protected:
+
+  using Base::derived;
+
+  MANIF_DEFAULT_CONSTRUCTOR(BundleTangentBase)
+
+public:
+
+  MANIF_TANGENT_ML_ASSIGN_OP(BundleTangentBase)
+
+  // Tangent common API
+
+  /**
+   * @brief Hat operator.
+   * @return An element of the Lie algebra.
+   */
+  LieAlg hat() const;
+
+  /**
+   * @brief Exponential operator.
+   * @return An element of the Lie Group.
+   */
+  LieGroup exp(OptJacobianRef J_m_t = {}) const;
+
+  /**
+   * @brief This function is deprecated.
+   * Please considere using
+   * @ref exp instead.
+   */
+  MANIF_DEPRECATED
+  LieGroup retract(OptJacobianRef J_m_t = {}) const;
+
+  /**
+   * @brief Get the right Jacobian.
+   */
+  Jacobian rjac() const;
+
+  /**
+   * @brief Get the left Jacobian.
+   */
+  Jacobian ljac() const;
+
+  /**
+   * @brief Get the inverse of the right Jacobian.
+   */
+  Jacobian rjacinv() const;
+
+  /**
+   * @brief Get the inverse of the right Jacobian.
+   */
+  Jacobian ljacinv() const;
+
+  /**
+   * @brief
+   */
+  Jacobian smallAdj() const;
+
+
+  // BundleTangent specific API
+
+  /**
+   * @brief Number of parts in the BundleTangent
+   */
+  static constexpr std::size_t BundleSize = IdxList::size();
+
+  /**
+   * @brief Access BundleTangent part as Map
+   * @tparam _Idx part index
+   */
+  template<int _Idx>
+  Eigen::Map<PartType<_Idx>> get();
```

Comment:
In the paper we speak of `block`s, but a function `block()` does not appear too explicit for me either.

I suppose from `BundleTangent<PartType<Idx>>::function()` we obtain the `Idx`-th tangent element?

And similarly that from `Bundle<PartType<Idx>>::function()` we obtain the `Idx`-th group element?

What about `element()`, `block()`, `part()`?

Reply:
OK I think I see it clearer now. 

If the syntax `BundleTangent::get<Idx>()` is correct, then I'd use `block<Idx>()`

Same for `Bundle::block<Idx>()`

Reply:
Whatever the decision, I think it's good to be consistent. So, if one part of the code uses `PartType`, then this should be `part<Idx>()`. Otherwise, use `BlockType` and `block<Idx>()`.

Reply:
`Block` as a meaning in `Eigen` so I'd rather use `element` to avoid confusion.

Reply:
Actually, the meaning of `block` in Eigen is quite similar than in our case, if that helps.

`element` is OK too.

Would it be `ElementType` too, instead of `PartType`?

Reply:
>Would it be ElementType too, instead of PartType?

I suppose so. Note that `PartType<Idx>` really only is a place holder for whatever underlying `manif` type (`SO2`, `RnTangent` etc...).

Reply:
OK I just said it in case `Part` had the same semantic meaning as `element`, otherwise it's not good to use the same word of course!

Reply:
BTW it would be interesting to see Examples V.B and V.C in the paper implemented with the new bundle!

Reply:
Thanks for the suggestions, I went with ```block<Idx>()``` here